### PR TITLE
make 0.7 work

### PR DESCRIPTION
**Pull request by SimonDanisch**
TODO
* rewrite broadcast
* get back 0.6 compat?!

### CODE REVIEW

File: src/abstractarray.jl [hunk 1]
```diff
@@ -190,7 +189,7 @@ Copyright (c) 2009-2016: Jeff Bezanson, Stefan Karpinski, Viral B. Shah, and oth
 
 https://github.com/JuliaLang/julia/contributors
 
-Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+Permission is hereby granted, free of charge, to any person obtaining a copyto of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copyto, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
```
Comment (jekbradbury) [inline src/abstractarray.jl]: over-eager find and replace

**Comment by SimonDanisch** [comment]
grafik
:)

**Comment by MikeInnes** [comment]
Looks like some issues cropped up on Julia master:

```
julia> using GPUArrays
[ Info: Precompiling module GPUArrays
┌ Warning: `convert(::Type{VersionNumber}, v::AbstractString)` is deprecated, use `VersionNumber(v)` instead.
│   caller = top-level scope
└ @ Core :0
WARNING: could not import Broadcast.broadcast_similar into GPUArrays
WARNING: importing deprecated binding Broadcast.broadcast_indices into GPUArrays.
ERROR: LoadError: UndefVarError: map_newindexer not defined
```

**Comment by vchuravy** [comment]
@SimonDanisch just fyi https://github.com/jrevels/MixedModeBroadcastAD.jl/pull/40

**Comment by SimonDanisch** [comment]
Ah great, was meaning to look for this :)

**Comment by MikeInnes** [comment]
The majority of tests are failing for me on this branch; any idea what might be going on? Do I need to check out some other packages perhaps?

**Comment by SimonDanisch** [comment]
No, seems just like a lot has changed on 0.7, besides the broadcast...

**Comment by MikeInnes** [comment]
Would be great to see this working now that 0.7 is stable; we'll need it to get the rest of the GPU ecosystem working.

**Comment by SimonDanisch** [comment]
Yes! Any help is appreciated :)

**Comment by Keno** [comment]
> get back 0.6 compat?!

I wouldn't bother trying to make this work on 0.6. Embrace 0.7, let femtocleaner clean out any cruft and move forward from there ;)

**Comment by Keno** [comment]
Femtocleaner is working on this PR, but ran into an error on one file, so you want me to look into that first: https://github.com/JuliaComputing/Deprecations.jl/issues/48

File: src/testsuite/testsuite.jl [hunk 2]
```diff
@@ -5,7 +5,19 @@ Abstract test suite that can be used for all packages inheriting from GPUArray
 """
 module TestSuite
 
-using Base.Test
+using GPUArrays
+using GPUArrays: mapidx, gpu_sub2ind
+
+if GPUArrays.is_v07
```
Comment (Keno) [inline src/testsuite/testsuite.jl]: FWIW, this kind of thing is not as good as a proper version check, since femtocleaner won't remove it.

**Comment by SimonDanisch** [comment]
closed in favor of https://github.com/JuliaGPU/GPUArrays.jl/pull/127#issuecomment-398693329

**Pull request closed without merging**